postgres/src/test/java/org/jdbi/v3/postgres/TestPostgresTypes.java

```diff
@@ -0,0 +1,232 @@
+        handle.useTransaction(h -> {
+            h.execute("drop table if exists postgres_custom_types");
+            h.execute("create table postgres_custom_types(id integer not null, foo text, bar text, created_on timestamp)");
+
+            // create custom type
+            h.execute("drop function if exists get_foo_bars()");
+            h.execute("drop function if exists get_foo_bar(integer)");
+            h.execute("drop function if exists insert_foo_bar(foo_bar_type)");
+            h.execute("drop function if exists insert_foo_bars(foo_bar_type[])");
+            h.execute("drop type if exists foo_bar_type");
+
+            h.execute("CREATE TYPE foo_bar_type AS (id integer, foo text, bar text);");
+
+            //create functions using custom types
+            h.execute("CREATE OR REPLACE FUNCTION get_foo_bars() RETURNS SETOF foo_bar_type AS \n"
+                    + "$$ \n"
+                    + "SELECT id, foo, bar FROM postgres_custom_types;\n"
+                    + "$$ LANGUAGE sql;");
+
+            h.execute("CREATE OR REPLACE FUNCTION get_foo_bar(aId integer) RETURNS foo_bar_type AS \n"
+                    + "$$ \n"
+                    + "SELECT id, foo, bar FROM postgres_custom_types WHERE id = aId;\n"
+                    + "$$ LANGUAGE sql;");
+
+            h.execute("CREATE OR REPLACE FUNCTION insert_foo_bar(aFooBar foo_bar_type) RETURNS void AS \n"
+                    + "$$\n"
+                    + "DECLARE\n"
+                    + "\n"
+                    + "BEGIN\n"
+                    + "INSERT INTO postgres_custom_types(id, foo, bar, created_on) VALUES(aFooBar.id, aFooBar.foo, aFooBar.bar, current_timestamp);"
+                    + "\n"
+                    + "END;\n"
+                    + "$$ LANGUAGE plpgsql;");
+
+            h.execute("CREATE OR REPLACE FUNCTION insert_foo_bars(aFooBars foo_bar_type[]) RETURNS void AS \n"
+                    + "$$\n"
+                    + "DECLARE\n"
+                    + "qFooBarType foo_bar_type;\n"
+                    + "BEGIN\n"
+                    + "FOREACH qFooBarType IN ARRAY aFooBars\n"
+                    + "LOOP \n"
+                    + "INSERT INTO postgres_custom_types(id, foo, bar, created_on) VALUES(qFooBarType.id, qFooBarType.foo, qFooBarType.bar, current_timestamp);"
+                    + "END LOOP;\n"
+                    + "END;\n"
+                    + "$$ LANGUAGE plpgsql;");
+
+            handle.execute("INSERT INTO postgres_custom_types(id, foo, bar, created_on) VALUES(1, 'foo1', 'bar1', current_timestamp)");
+            handle.execute("INSERT INTO postgres_custom_types(id, foo, bar, created_on) VALUES(2, 'foo2', 'bar2', current_timestamp)");
+        });
+    }
+
+    @After
+    public void after() {
+        handle.close();
+    }
+
+    @Test
+    public void testReadViaFluentAPI() {
+        FooBarPGType result = (FooBarPGType) handle.createQuery("SELECT get_foo_bar(1)")
+                .map(new PGObjectColumnMapper())
```

Comment:
Would be better if we could use `mapTo(FooBarPGType.class)` instead of having to cast. I think the column mapper needs to be a column mapper factory for that.